### PR TITLE
=str #19958 Throttle does not work for several flows.

### PR DESCRIPTION
**Pull request by agolubev**
Added tests on java and scala. Java test mimics what's in example.
Seems like throttle itself is Ok. Perhaps issue is in configuration.
Ref #19958

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

File: akka-stream-tests/src/test/java/akka/stream/javadsl/FlowThrottleTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            return Supervision.restart();
+        };
+
+        final Materializer mat1 = ActorMaterializer.create(ActorMaterializerSettings.create(system).withSupervisionStrategy(decider), system);
+        final Materializer mat2 = ActorMaterializer.create(ActorMaterializerSettings.create(system).withSupervisionStrategy(decider), system);
+
+        final String name1 = "Pipe 1: ";
+        final String name2 = "Pipe 2: ";
+
+        Source<Object, NotUsed>  pipe1 = createPipeLine(name1);
```
Comment (drewhk) [inline akka-stream-tests/src/test/java/akka/stream/javadsl/FlowThrottleTest.java]: this and line 50 creates two different throttle instances so it does not test reusability.

**Comment by drewhk** [comment]
I have a branch on various throttle fixes, so please do not take tickets on throttle for now.

**Comment by agolubev** [comment]
>I have a branch on various throttle fixes, so please do not take tickets on throttle for now.

OK

**Comment by ktoso** [comment]
I'll mark it stashed then.

**Comment by agolubev** [comment]
@drewhk, is the example in ticket attachment useful to be in tests? or  you already added it?

**Comment by drewhk** [comment]
I will add your tests

**Comment by agolubev** [comment]
Thanks!

**Comment by patriknw** [comment]
@drewhk how does it relate to your other throttle PR?

**Comment by drewhk** [comment]
Once mine PR is merged we should rebuild this (so the test runs against the new impl) and then merge

**Comment by drewhk** [comment]
unstashed it, and marked as pick-next

**Comment by patriknw** [comment]
thanks for the update

**Comment by drewhk** [comment]
@agolubev can you maybe rebase or should I do it myself?

**Comment by drewhk** [comment]
never mind, I port it manually.

**Comment by drewhk** [comment]
superseded by: https://github.com/akka/akka/pull/20191

**Pull request closed without merging**